File: Easy/66_Plus_One.py
class Solution(object):
    def plusOne(self, digits):
        carry = False
        for x in range(len(digits) - 1, -1, -1):
            digits[x] += 1
            if(digits[x] != 10):
                return digits
            if(digits[x] == 10):
                digits[x] = 0
            if(x == 0):
                digits[x] = 1
                digits.append(0)
        return digits

File: EasyJS/66_Plus_One.js
var plusOne = function (digits) {
  //   let number = parseInt(digits.join());
  //   number += 1;
  //   return num.toString().split("").map(Number);
  let carry = true;
  for (let i = digits.length - 1; i >= 0; i--) {
    if (carry) {
      if (digits[i] === 9) {
        digits[i] = 0;
      } else {
        digits[i] += 1;
        carry = false;
      }
    } else {
      break;
    }
  }
  if (carry) {
    digits.unshift(1);
  }
  return digits;
};

console.log(plusOne([1, 2, 3]));
